### ISSUE TITLE
[question] how to have a abstract class/interface as a property/field without knowing the type before

### ISSUE THREAD

**Issue by TMaddox** [dahomey-technologies/Dahomey.Cbor#20]
I know this is possible https://github.com/dahomey-technologies/Dahomey.Cbor/issues/11, but have you got any any idea on how I could use this if I do not know the type?
```
public class Tree
{
        public Fruit { get; set; }
        ...
}

public abstract class Fruit { ... }

public class Apple : Fruit { ... }
public class Orange : Fruit { ... }
```

**Comment by mcatanzariti** [comment]
The implementation of polymorphism in Dahomey.Cbor is based on a special field in which a special value is associated with an inherited class. The special field/value is called discriminator.

By default the discriminator field is named "_t" and the discriminator values are configured

1. Either by using the attribute CborDiscriminatorAttribute:

```csharp
[CborDiscriminator("Apple")]
public class Apple : Fruit { ... }

[CborDiscriminator("Orange")]
public class Orange : Fruit { ... }
```

Moreover , in this option, you need to register the assembly to allow the framework to discover the classed flagged by the CborDiscriminatorAttribute:

```csharp
CborOptions.Default.Registry.DefaultDiscriminatorConvention.RegisterAssembly(typeof(Fruit ).Assembly);
```

2. Or by API:

```csharp
options.Registry.ObjectMappingRegistry.Register<Apple>(om => om.AutoMap().SetDiscriminator("Apple");
options.Registry.ObjectMappingRegistry.Register<Orange>(om => om.AutoMap().SetDiscriminator("Orange");
```

**Comment by TMaddox** [comment]
I tried both examples and I also added the [CborDiscriminator("Name")] attribute on all classes implementing the Fruit class, but it still throws: 
`CborException: A CreatorMapping should be defined for interfaces or abstract classes`

@mcatanzariti I am mentioning you by name because I do not have the permission to reopen this issue, but I feel like I should ask here.

**Comment by mcatanzariti** [comment]
Did you try with the last official version?

**Comment by TMaddox** [comment]
yes with 1.3.2, I have broken it down to tree classes while testing and found out the following:
```
    public class Tree
    {
        [CborProperty("age")]
        public readonly int _age;
        
        [CborProperty("fruit")]
        public Fruit _fruit;

        [CborConstructor]
        public Tree(int age, Fruit fruit)
        {
            _age = age;
            _fruit = fruit;
        }
    }

    public abstract class Fruit
    {
        public string id2 = "myid"; // if I remove this it won't throw
    }

    [CborDiscriminator("Apple")]
    public class Apple : Fruit
    {
        [CborProperty("type")]
        public string type = "I am an apple";
    }

    [CborDiscriminator("Orange")]
    public class Orange : Fruit
    {
        [CborProperty("type")]
        public string type = "I am an orange";
    }
```

**Comment by TMaddox** [comment]
further, serializing the above example will output `{"age": 10, "fruit": {}, "id": "123"}`, with fruit being empty, I have tried it with const fields and normal ones in combination with a constructor.

**Comment by mcatanzariti** [comment]
Reproduced

**Comment by mcatanzariti** [comment]
fix in [1.4.0](https://github.com/dahomey-technologies/Dahomey.Cbor/releases/tag/1.4.0)
